### DISCUSSION TITLE
Installing and configurating Firebase Phone Authentification

### DISCUSSION THREAD

**Discussion by Soufiane-lab** [invertase/react-native-firebase#4297] [Q&A] [answered]
Hello,

I hade an issue working with Phone authentification. I have successfully done the firebase phone installation, but when I want to working with `signInWithPhoneNumber ` I get errors in phone confirmation code.

is a` .confirm()` a function ? and how we can confirm the code separately from the `signIn `method explained in https://rnfirebase.io/auth/phone-auth. It's seems like working on this example but for complex UI it doesn't work.

Thank you in advance for any help or advice

**Comment by mikehardy** [comment, marked as answer]
> It's seems like working on this example but for complex UI it doesn't work.

So, the module works?

Note that on Android the codes may be auto-consumed by the underlying libraries, meaning that you will never actually see the code yourself, rather you will listen to the user auth status changed listener and advance your app state based on the info provided to you in the callback

https://rnfirebase.io/auth/usage#listening-to-authentication-state

https://rnfirebase.io/reference/auth#onAuthStateChanged

I found the V5 documentation to be more complete in this area but haven't had time to port it forward to v6+ PRs to do so very happily accepted (there's an edit button top right of every docs page to propose a docs change)

https://v5.rnfirebase.io/docs/v5.x.x/auth/phone-auth#3.-[Android]-Handle-Auto-Verification

**Reply by Soufiane-lab** [reply]
Yes, the module is working sometimes, but for example, when you use multiple screens you will easily get lost with hooks and all the functions don't work properly.

I think the v6 is more complete than this one, I will check and see what i can do to improve the the documentation of v6. the only problem is that with thenew version we don't use `firebase...` we use the functions directly. In the latest documentation you didn't mentioned the `verifyPhoneNumber ` method, is it working properly on v6 ?

**Reply by mikehardy** [reply]
To the best of my knowledge all the auth APIs on v6 work. How you reference them is a matter of how you import them? They could be named or scoped almost anything depending on you do that. Something like redux or resub could help you manage state.

**Reply by Soufiane-lab** [reply]
Ok, thank you, the state changing between screens is sometimes difficult and spatially when using the different types of authentication.
I will try to understand it and if I have some times left I can contribute by improving the documentation. I think the react-native-firebase module is very useful but I think when we understand it it will better if you can simplify the documentation
